internal/core/window.rs

```diff
@@ -110,6 +110,12 @@ pub trait WindowAdapter {
     /// Return the size of the Window on the screen
     fn size(&self) -> PhysicalSize;
 
+    /// Return the inset of the safe area of the Window in physical pixels.
+    /// This is necessary to avoid overlapping system UI such as notches or system bars.
+    fn safe_area_inset(&self) -> PhysicalInset {
```

Comment:
Maybe we do not need this function and have the platform send the `SafeAreaChanged` event instead. Would that work?